client/src/library/hooks/use-delete-progress.ts

```diff
@@ -0,0 +1,50 @@
+import { getLibraryService } from "@/domains/game/library-service";
+import { getLocalRepository, getRemoteAPIRepository } from "@/repositories";
+import { useState } from "react";
+
+export const useDeleteProgress = () => {
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [error, setError] = useState<string | null>(null);
```

Comment:
c'est stylé d'avoir géré l'état de loading/error ! On a tanstack-query dans la codebase qui permet de gérer tout ça pour nous. Dans ton cas c'est `useMutation` qui va être utile (comme on le fait dans `auth-modal-form.tsx`)

ça ressemblerait à quelque chose comme ça :

```suggestion
  const {
    mutateAsync: deleteProgress,
    isPending,
    isError,
  } = useMutation({
    mutationFn: async (progressKey: string) => {
      const localRepository = getLocalRepository();
      await localRepository.deleteStoryProgresses([progressKey]);
    },
    onError: () => {
      // TODO: handle error, show toast...
    },
  });
```